distributed/worker.py

```diff
@@ -2361,6 +2375,8 @@ async def gather_dep(
 
                 if response["status"] == "busy":
                     self.log.append(("busy-gather", worker, to_gather_keys))
+                    self.busy_workers.add(worker)
+                    self.busy_workers_log.add(worker)
```

Comment:
I would prefer you using the `Worker.log` and test for the `busy-gather` event instead of introducing another `set` which is simply used for testing